spotinst/resource_spotinst_ocean_spark_virtual_node_group_test.go

```diff
@@ -0,0 +1,185 @@
+package spotinst
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/spotinst/spotinst-sdk-go/service/ocean/spark"
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/commons"
+)
+
+const (
+	testOceanSparkClusterID = "osc-cc4b42b4"
+	testOceanDedicatedVngID = "ols-d04cb833"
+)
+
+var oceanSparkClusterID = getOceanSparkClusterID() // NOTE: This needs to be an existing ocean cluster
+var oceanSparkVngID = getOceanSparkVngID()         // NOTE: This needs to be an existing ocean VNG
+
+func init() {
+	resource.AddTestSweepers("spotinst_ocean_spark_virtual_node_group", &resource.Sweeper{
+		Name: "spotinst_ocean_spark_virtual_node_group",
+		F:    testSweepOceanSparkVng,
+	})
+}
+
+func getOceanSparkClusterID() string {
+	// Prefer environment variable
+	oceanSparkClusterID := os.Getenv("TEST_ACC_OCEAN_SPARK_CLUSTER_ID")
+	if oceanSparkClusterID == "" {
+		// Default to hardcoded ID
+		oceanSparkClusterID = testOceanSparkClusterID
+	}
+
+	return oceanSparkClusterID
+}
+
+func getOceanSparkVngID() string {
+	// Prefer environment variable
+	oceanDedicatedVngID := os.Getenv("TEST_ACC_OCEAN_SPARK_DEDICATED_VNG_ID")
+	if oceanDedicatedVngID == "" {
+		// Default to hardcoded ID
+		oceanDedicatedVngID = testOceanDedicatedVngID
+	}
+
+	return oceanDedicatedVngID
+}
+
+func testSweepOceanSparkVng(_ string) error {
+	client, err := getProviderClient("aws")
+	if err != nil {
+		return fmt.Errorf("error getting client: %v", err)
+	}
+
+	conn := client.(*Client).ocean.Spark()
+	input := &spark.ListVngsInput{ClusterID: spotinst.String(oceanSparkClusterID)}
+	if resp, err := conn.ListVirtualNodeGroups(context.Background(), input); err != nil {
+		return fmt.Errorf("error getting VNGs to sweep")
+	} else {
+		if len(resp.VirtualNodeGroups) == 0 {
+			log.Printf("[INFO] No VNGs to sweep")
+		}
+		for _, vng := range resp.VirtualNodeGroups {
+			// Our test clusters should have a controller cluster ID starting with "tf-test-acc-"
+			if strings.Compare(spotinst.StringValue(vng.VngID), oceanSparkVngID) == 0 {
+				if _, err := conn.DetachVirtualNodeGroup(context.Background(), &spark.DetachVngInput{ClusterID: spotinst.String(oceanSparkClusterID), VngID: vng.VngID}); err != nil {
+					return fmt.Errorf("unable to detach VNG %v in sweep", spotinst.StringValue(vng.VngID))
```

Comment:
include the underlying error?
`return fmt.Errorf("unable to detach VNG %v in sweep, %w", spotinst.StringValue(vng.VngID), err)`